lib/migrations/migrate/Migrator.js

```diff
@@ -80,12 +80,16 @@ class Migrator {
 
     const transactionForAll =
       !this.config.disableTransactions &&
-      !migrations.some((migration) => {
-        const migrationContents = this.config.migrationSource.getMigration(
-          migration
-        );
-        return !this._useTransaction(migrationContents);
-      });
+      !(
+        await Promise.all(
+          migrations.map(async (migration) => {
+            const migrationContents = await this.config.migrationSource.getMigration(
+              migration
+            );
+            return !this._useTransaction(migrationContents);
+          })
+        )
+      ).some((e) => e);
```

Comment:
Maybe a slightly more descriptive name could be used? from my understanding the value you'll have there is `isTransactionUsed` or something along these lines